Modules/Video/Core/test/itkImageToVideoFilterTest.cxx

```diff
@@ -0,0 +1,116 @@
+
+#include <string>
+
+#include "itkImage.h"
+#include "itkVideoStream.h"
+
+#include "itkImageFileReader.h"
+#include "itkImageFileWriter.h"
+#include "itkImageSliceConstIteratorWithIndex.h"
+#include "itkImageLinearConstIteratorWithIndex.h"
+#include "itkTestingMacros.h"
+
+int
+itkImageToVideoFilterTest(int argc, char * argv[])
+{
+  if (argc < 3)
+  {
+    std::cerr << "Usage: " << argv[0];
+    std::cerr << " inputFile outputFile";
+    std::cerr << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  using PixelType = float;
+  const unsigned int Dimension = 3;
+
+  using ImageType = itk::Image<PixelType, Dimension>;
+
+  // Get 3D image to represent a temporal dataset of 2D frames
+  using ReaderType = itk::ImageFileReader<ImageType>;
+  ReaderType::Pointer reader = ReaderType::New();
+  reader->SetFileName(argv[1]);
+  reader->Update();
+
+  using VideoFilterType = itk::ImageToVideoFilter<ImageType>;
+  using VideoType = typename VideoFilterType::OutputVideoStreamType;
+  VideoFilterType::Pointer videoFilter = VideoFilterType::New();
+  videoFilter->SetInput(reader->GetOutput());
+  // Arbitrarily set 0th axis as temporal dimension to split frames
+  itk::IndexValueType frameAxis = 0;
+  videoFilter->SetFrameAxis(frameAxis);
+
+  ITK_TRY_EXPECT_NO_EXCEPTION(videoFilter->Update());
+
+  auto videoOutput = videoFilter->GetOutput();
+  auto imageRegion = reader->GetOutput()->GetLargestPossibleRegion();
+
+  // Verify start frame and frame duration in output match size of designated temporal axis in input
+  ITK_TEST_EXPECT_EQUAL(
+    static_cast<itk::IndexValueType>(videoOutput->GetLargestPossibleTemporalRegion().GetFrameStart()),
+    static_cast<itk::IndexValueType>(reader->GetOutput()->GetLargestPossibleRegion().GetIndex(frameAxis)));
+  ITK_TEST_EXPECT_EQUAL(
+    static_cast<itk::IndexValueType>(videoOutput->GetLargestPossibleTemporalRegion().GetFrameDuration()),
+    static_cast<itk::IndexValueType>(reader->GetOutput()->GetLargestPossibleRegion().GetSize(frameAxis)));
+
+  // Verify spatial dimensions, spacing, and origin in output frames match size of non-temporal axes in input
+  for (itk::IndexValueType idx = 1; idx < ImageType::ImageDimension; idx++)
+  {
+    ITK_TEST_EXPECT_EQUAL(videoOutput->GetFrame(0)->GetLargestPossibleRegion().GetSize(idx - 1),
+                          reader->GetOutput()->GetLargestPossibleRegion().GetSize(idx));
+    ITK_TEST_EXPECT_EQUAL(videoOutput->GetFrame(0)->GetSpacing()[idx - 1], reader->GetOutput()->GetSpacing()[idx]);
+    ITK_TEST_EXPECT_EQUAL(videoOutput->GetFrame(0)->GetOrigin()[idx - 1], reader->GetOutput()->GetOrigin()[idx]);
+  }
+
+  // Iterate over 3D input + video output to verify pixel data matches across each slice/frame
+  using ImageIteratorType = typename itk::ImageSliceConstIteratorWithIndex<ImageType>;
+  ImageIteratorType it(reader->GetOutput(), reader->GetOutput()->GetLargestPossibleRegion());
+  it.SetFirstDirection(2);  // fastest moving in original image
```

Comment:
Are you sure you are not inverting index order here? Isn't 0=i, 1=j, 2=k? Given the nested while loops which come later, a mistake here will only make the iteration slower. Since axis 0 is frame splitting, I expect axis 1 to be faster changing than axis 2.

Reply:
Per above discussion you are correct. I will flip the directions to reflect that axis 1 is moving faster than axis 2.